### DISCUSSION TITLE
Support for libgpiod

### DISCUSSION THREAD

**Discussion by waneby** [labgrid-project/labgrid#1372] [Ideas]
Hi

I am currently working with Labgrid on a scholar project to investigate the possibilities of using it to control embedded systems for development. I think it is a great tool and offers a lot of flexibility.
 
But when I tried to control GPIO pins with it, to power on/off my target device, some problems occurred. My current operating system will remove support for sysfs interfaces in the near future. As far as my understanding goes, the whole Linux kernel moved away from using the sysfs user space implementation and marked it deprecated.
 
https://www.kernel.org/doc/html/latest/driver-api/gpio/legacy.html#sysfs-interface-for-userspace-optional
 
I was trying to write my own labgrid resource implementation for the new interface using libgpiod (the new descriptor-based interface), but I got to the limit of my abilities to write python code…
 
So I was wondering if there are any plans to support the new libgpiod interface, instead of sysfs, to control GPIO pins?
 
https://www.kernel.org/doc/html/latest/driver-api/gpio/using-gpio.html
(libgpiod is mentioned on the bottom)
 
I think it would be a neat addition to have labgrid support for the new Linux GPIO interface. Because of that, I post this in the ideas section :)

Best Wishes

**Comment by Bastian-Krause** [comment]
See #468 for previous discussions.

**Reply by waneby** [reply]
Well my bad I did not dig deep enough. Thanks for the reply.
